tokio-stream/tests/stream_chunks_remainder.rs

```diff
@@ -0,0 +1,26 @@
+#![warn(rust_2018_idioms)]
+#![cfg(all(feature = "time", not(target_os = "wasi")))] // Wasi does not support panic recovery
+#![cfg(panic = "unwind")]
+
+use futures::FutureExt;
+use std::error::Error;
+use tokio::time::Duration;
+use tokio_stream::{self as stream, StreamExt};
+
+#[tokio::test]
+async fn stream_chunks_remainder() -> Result<(), Box<dyn Error>> {
+    let inner = stream::iter([1, 2, 3, 4]).chain(stream::pending());
+    tokio::pin!(inner);
+
+    let chunked = (&mut inner).chunks_timeout(10, Duration::from_millis(20));
+    tokio::pin!(chunked);
+
+    tokio::select! {
+        Some(_chunk) = chunked.next() => {}
+        _ = tokio::time::sleep(Duration::from_millis(3)) => {}
+    }
```

Comment:
Could we poll the `chunked` manually? So that we can simplify the test by removing the timer and `tokio::select`.